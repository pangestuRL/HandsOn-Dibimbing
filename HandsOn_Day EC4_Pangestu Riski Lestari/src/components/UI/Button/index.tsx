import React from 'react';
import { ButtonProps } from './Button.interface';

const Button = (props: ButtonProps) => {
  const { children, onClick } = props;

  const handleClick = () => {
    if (onClick) {
      onClick();
    }
  };
  return <button onClick={handleClick}>{children}</button>;
};

export default Button;
