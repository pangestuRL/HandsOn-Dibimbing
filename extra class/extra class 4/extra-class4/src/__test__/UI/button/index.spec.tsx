import Button from "@/components/UI/Button";
import { render } from "@testing-library/react";

describe('Button component test', () => {
    it ('should handle button render', () => {
        const { container } = render(<Button>Test</Button>);
        expect (container).toMatchSnapshot();
    });
});