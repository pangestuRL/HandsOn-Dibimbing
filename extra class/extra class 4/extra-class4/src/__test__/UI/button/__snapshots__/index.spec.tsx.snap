// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should handle button render 1`] = `
<div>
  <button>
    Test
  </button>
</div>
`;
